src/controllers/product.controller.js

```diff
@@ -0,0 +1,61 @@
+import jwt from "jsonwebtoken";
+import models from "../database/models";
+
+export const addProduct = async (req, res) => {
+  try {
+    const authHeader = req.headers.authorization;
+    if (!authHeader) {
+      return res.status(401).json({ message: "Authorization header missing" });
+    }
+    const token = authHeader.split(" ")[1];
+    const decodedToken = jwt.verify(token, process.env.SECRET_KEY);
+    const sellerId = decodedToken.userId;
+
+    const user = await models.User.findByPk(sellerId);
+    if (!user) {
+      return res.status(400).json({ message: "Seller not found" });
+    }
+    if (
+      sellerId !== decodedToken.userId ||
+      decodedToken.userRole !== "vendor"
+    ) {
+      return res.status(403).json({ message: "Unauthorized access" });
+    }
+
+    const {
+      name,
+      price,
+      quantity,
+      available,
+      category,
+      bonus,
+      expiryDate,
+      ec,
+    } = req.body;
+    const images = req.body.images || [];
+    const existingProduct = await models.Product.findOne({ where: { name } });
+    if (existingProduct) {
+      return res.status(409).json({
+        message: "Product already exists you can update that product instead",
+      });
+    }
```

Comment:
It's not a good practice to mix validation and create a product controller, better to handle it in middleware.